CartoDB/cartodb
# Allows exporting static maps with Google Basemaps

This PR extends our current static map export and allows exporting google maps basemaps.

![untitled map 1 44](https://cloud.githubusercontent.com/assets/4933/24455806/357208e4-1491-11e7-9814-cce4e1d18b1d.png)

### Work done in this PR: 

- [x] Export defaults / non-styled Google Maps basemaps.
- [x] Export styled Google Maps basemaps.
- [x] Allow panning the basemap when the image export widget is active.
- [x] Apply Google Maps limits in the export image form and warn the users.
  - [x] For regular users.
  - [x] For Google Maps Premium users.
- [x] Add attribution to the exported image (for regular basemaps and GMaps)
- [x] Add CARTO logo to the exported image.
- [x] Remove menu in the export image form header. 

### How to test this

- Enable the `static-maps` feature flag and then simply create a map (with or without geometries) and export it. 

- Enable the Google Maps API in your user.

- Change the basemap and export it again. Use GMaps or default ones. Check that the attribution and our logo appears in the image. 

- Check that with Google Basemaps the max dimensions of the image are these: 

  - **Default**: 640x640px
  - **Premium plan**: 2048x2048px (those that use a `&signature=` in the `google_maps_key`)
